memdx/pendingop_multi.go

```diff
@@ -0,0 +1,35 @@
+package memdx
+
+import "sync"
+
+type multiPendingOp struct {
+	ops       []PendingOp
+	lock      sync.Mutex
+	cancelErr error
+}
+
+func (op *multiPendingOp) Cancel(err error) {
+	op.lock.Lock()
+	op.cancelErr = err
+
+	ops := make([]PendingOp, len(op.ops))
```

Comment:
Just a performance tidbit, but `cancelErr != nil` is technically a locking mechanism for the ops (you can't add ops after its non-nil), so you can probably elide the copy here, knowing that the ops list cannot change.